etc/cmake/options.cmake

```diff
@@ -127,3 +127,8 @@ option(OTBR_DNS_UPSTREAM_QUERY "Allow sending DNS queries to upstream" OFF)
 if (OTBR_DNS_UPSTREAM_QUERY)
     target_compile_definitions(otbr-config INTERFACE OTBR_ENABLE_DNS_UPSTREAM_QUERY=1)
 endif()
+
+option(OTBR_PUBLISH_MESHCOP_ID_AS_EXT_ADDRESS "Publish the MeshCoP mDNS 'id' TXT entry as the Extended MAC Address, enable this feature only when 'id' is not set via dbus API" OFF)
```

Comment:
Can we default to ON and allow override via API and/or DBus?

Reply:
I am okay to default ON.

But to allow override via API or d-bus, there can be race conditions that different IDs are advertised for the same device: if the API is called after the default MeshCoP mDNS service has been advertised. It's even worse to advertise a wrong ID than missing an ID for a short while. Thoughts?

Reply:
currently the mobile platform will fallback to `xa`, if so, it will have race condition whether or not the `id` is published.

Reply:
My concern here is that a default OTBR build will not include the `id` field, which can cause issues for those using mobile OS APIs. Can we have a build switch that configures how the `id` field is set?

Reply:
Even the Extended MAC Address version here is not a preferred generation of the ID, making it default ON is suggesting a non-preferred solution. If we want a default ID implementation, we should follow https://threadgroup.atlassian.net/browse/SPEC-1020 to randomly generate it and save in the persistent settings. This was not added in ot-br-posix because we think vendor BRs already have a kind of ID in their ecosystem. But it does seem to be the case?

What about highlighting the Thread BR requirements in the mobile OS API guide doc?